Source/Private/ASIGListAdapterBasedDataSource.m

```diff
@@ -139,6 +154,32 @@ - (void)collectionNode:(ASCollectionNode *)collectionNode willBeginBatchFetchWit
   [ctrl beginBatchFetchWithContext:context];
 }
 
+- (void)collectionNode:(ASCollectionNode *)collectionNode willDisplayItemWithNode:(ASCellNode *)node
+{
+  NSIndexPath *indexPath = [collectionNode.view indexPathForNode:node];
+  UIView *contentView = node.view.superview;
+  UICollectionViewCell *cell = contentView.superview;
+
+  if (cell == nil || indexPath == nil) {
+    return;
+  }
+
+  [self.delegate collectionView:collectionNode.view willDisplayCell:cell forItemAtIndexPath:indexPath];
+}
+
+- (void)collectionNode:(ASCollectionNode *)collectionNode didEndDisplayingItemWithNode:(ASCellNode *)node
+{
+  NSIndexPath *indexPath = [collectionNode.view indexPathForNode:node];
+  UIView *contentView = node.view.superview;
+  UICollectionViewCell *cell = contentView.superview;
+
+  if (cell == nil || indexPath == nil) {
+    return;
+  }
+
+  [self.delegate collectionView:collectionNode.view didEndDisplayingCell:cell forItemAtIndexPath:indexPath];
```

Comment:
Actually, since we conform to `ASCollectionDelegateInterop` we should be already bridging these. See lines ~190 below, where we pass these calls directly through to the delegate.

Can you put breakpoints in there or investigate why we're not seeing them?

Reply:
So, I've checked that and it never fires the delegate because don't pass this checking:
```obj-c
    if (node.shouldUseUIKitCell) {
      [(id <ASCollectionDelegateInterop>)_asyncDelegate collectionView:collectionView willDisplayCell:rawCell forItemAtIndexPath:indexPath];
    }
```
Here: https://github.com/TextureGroup/Texture/blob/master/Source/ASCollectionView.mm#L1183

I am trying to understand the `shouldUseUIKitCell` and why do we need the check here to fire that.

Reply:
If I comment out this check it works, Also I noticed that this will be always `NO` and will never fire the delegate because it is a read-only property, it is only "YES" if it is an `ASWrapperCellNode`.

Here it was checking:
https://github.com/TextureGroup/Texture/blob/550da242b7852563cbcbf409cb9e3373e626304a/Source/ASCollectionView.mm#L1100

It was first introduced here:
https://github.com/TextureGroup/Texture/commit/c6e3dd7a5d8bb0dc4dd26b96f96bf96946ade806#diff-8b82b1556d1293f6f9789beb236c9271R1167

Simplified here:
https://github.com/TextureGroup/Texture/commit/63efdbde8f958fcf5cf09a8df19e96cb90843a54#diff-8b82b1556d1293f6f9789beb236c9271R1169

For `ASWrapperCellNode` I have also found this:
https://github.com/TextureGroup/Texture/blob/master/Source/ASCellNode.mm#L387-L393

and there is a comment 
>// TODO: Consider if other calls, such as willDisplayCell, should be bridged to this class.

Reply:
Looks like @hebertialmeida is pointing out the problem, do this issue has any update?

Manual modifying ASIGListAdapterBasedDataSource.m isn't a nice solution for now as it is a private class. So it will be great if this issue get sort out and merge to master soon.

Thanks guys for the help.

Reply:
@MiguelChiang Still waiting someone get back to me.

Reply:
@Adlai-Holler  can you please take a look when you get a chance?